external/fv3fit/tests/data/test_windowed_zarr.py

```diff
@@ -0,0 +1,133 @@
+from fv3fit.data import WindowedZarrLoader, VariableConfig
+from fv3fit.data.tfdataset import get_n_windows
+import tempfile
+import xarray as xr
+import numpy as np
+import pytest
+import contextlib
+
+NX, NY, NZ, NT = 5, 5, 8, 40
+
+
+@contextlib.contextmanager
+def temporary_zarr_path():
+    dir = tempfile.TemporaryDirectory()
+
+    ds = xr.Dataset(
+        data_vars={
+            "a": xr.DataArray(
+                np.random.randn(NT, NX, NY, NZ), dims=["time", "x", "y", "z"]
+            ),
+            "a_sfc": xr.DataArray(np.random.randn(NT, NX, NY), dims=["time", "x", "y"]),
+            "b": xr.DataArray(
+                np.random.randn(NT, NX, NY, NZ), dims=["time", "x", "y", "z"]
+            ),
+            "c": xr.DataArray(
+                np.random.randn(NT, NX, NY, NZ), dims=["time", "x", "y", "z"]
+            ),
+        }
+    )
+    ds.to_zarr(dir.name)
+    yield dir.name
+
+
+@pytest.mark.parametrize(
+    "variable_names",
+    [pytest.param(["a"], id="one_var"), pytest.param(["b", "c"], id="two_vars")],
+)
+def test_loader_gets_requested_variables(variable_names: str):
+    with temporary_zarr_path() as data_path:
+        loader = WindowedZarrLoader(
+            data_path=data_path,
+            window_size=10,
+            unstacked_dims=["time", "z"],
+            default_variable_config=VariableConfig(times="window"),
+            variable_configs={},
+        )
+        dataset = loader.get_data(
+            local_download_path=None, variable_names=variable_names
+        )
+        item = next(iter(dataset))
+        assert set(item.keys()) == set(variable_names)
+
+
+def test_loader_stacks_default_config():
```

Comment:
Because the call signature for `records` is basically the entire init signature of WindowedZarrLoader, I'll leave it as is. Otherwise I'll have to duplicate some production code (opening the zarr store) in this test.